plugins/catalog/src/hooks/useEntities.ts

```diff
@@ -40,6 +40,88 @@ type UseEntities = {
   selectTypeFilter: (id: string) => void;
 };
 
+type EntityFilterGroupOutput<T> = {
+  selectItems: (items: string[]) => void;
+  filteredItems: T[];
+  states: OutputState;
+};
+
+type OutputState = { [key: string]: { isSelected: boolean; count: number } };
+
+type FilterDefinition<T> = {
+  [key: string]: {
+    isSelected: boolean;
```

Comment:
Not sure that this can be an input here inside the structure. We'll have to memoize somehow over the set of selected things, and we want the hook to help out with the state management for us.